### DISCUSSION TITLE
Annotation of the RTSP stream and VOUT using Eazy AI library does not work

### DISCUSSION THREAD

**Discussion by lipo5476** [AxisCommunications/discussions#31] [General] [answered]
Hi all,

We are developing application for AXIS M4216-V camera with Ambarella CV25 chip, it is using Ambarella SDK version 3.0 and Eazy AI library, it processes the input camera stream and annotates the stream with face detection results. Everything is running well on different vendor cameras with same Amba chip and SDK. However, we have issue on AXIS camera, see below:

When we tried to run the application with Eazy AI library it fails when calling ea_display_new function both with stream and VOUT as input parameter:

We are getting an error from display = ea_display_new(EA_DISPLAY_STREAM, options.stream_id, EA_DISPLAY_BBOX_TEXTBOX, NULL); function, see below the output of our application:

[Error]: ea_env()->valid_stream_map & (1 << stream_id) failed
[Trace] file /home/lipo/dev/Amba/cv2x_linux_sdk_3.0/ambarella/packages/eazyai/src/display_stream.c.
function: display_stream_new: line 1189
[Error]: display->stream != NULL failed
[Trace] file /home/lipo/dev/Amba/cv2x_linux_sdk_3.0/ambarella/packages/eazyai/src/ea_display.c.
function: ea_display_new: line 129
Aborted (core dumped)

When we tried HDMI output like this display = ea_display_new(EA_DISPLAY_VOUT, EA_DISPLAY_ANALOG_VOUT, EA_DISPLAY_BBOX_TEXTBOX, NULL);:

SMARTFB_ERROR:: smartfb_init(1379): Open /dev/fb1 fail!
Open fb error: Success
[Error]: smartfb_init(&smartfb) failed, return -1
[Trace] file /home/lipo/dev/Amba/cv2x_linux_sdk_3.0/ambarella/packages/eazyai/src/display_vout.c.
function: display_vout_new: line 134
[Error]: display->vout != NULL failed
[Trace] file /home/lipo/dev/Amba/cv2x_linux_sdk_3.0/ambarella/packages/eazyai/src/ea_display.c.
function: ea_display_new: line 149
Aborted (core dumped)

Both RTSP and HDMI are turned on and running on the camera.
Could you please help us with this issue so we are able to annotate the RTSP stream or VOUT.

Thank you in advance
Marek

**Comment by mattias-kindborg-at-work** [comment]
Hi @lipo5476! A couple of questions:

1. Are you working towards an implementation where you display bounding boxes around the faces, or is the end goal something else?
2. It would potentially help us if we knew the manufacturer and model of devices where your code successfully ran. Could you please tell us more about these devices?

**Reply by lipo5476** [reply]
Hi @mattias-kindborg-at-work, thanks for response see below my answers:
1. It is not the main goal, but we need to be able to annotate the output RTSP and/or HDMI stream for demonstration purposes.
2. The annotation works on Ambarella dev board and on the different cameras from Taiwan using the Ambarella CV22/25 chip and the same Amba SDK which is running on the mentioned AXIS camera.

I believe the difference is how the stream is handled on different cameras

**Reply by mattias-kindborg-at-work** [reply]
Hi @lipo5476 and sorry for the late reply.

Although some of our products are built on Ambarella, the API of Ambarella is not something we officially support on our products. The same goes for Eazy AI. Different device models are built using different hardware, and the APIs we support are the same indifferent of the device model used. That's why Axis features [VAPIX](https://www.axis.com/vapix-library/) and [ACAP](https://axiscommunications.github.io/acap-documentation/). Please see if any of these APIs can help you out in your use case.

**Comment by m-chichikalov** [comment]
@lipo5476 , may I ask, where did you get *eazyai* header and library to compile against it? I cannot find in any Docker image that Axis provides.

@mattias-kindborg-at-work maybe you can help me to point to some examples of using this library?

**Reply by mattias-kindborg-at-work** [reply, marked as answer]
Hi @m-chichikalov and sorry for the late reply.

Eazy AI and Ambrella APIs are not officially supported by Axis. The APIs we support for Axis devices are [VAPIX](https://www.axis.com/vapix-library/) and [ACAP](https://axiscommunications.github.io/acap-documentation/).
